modules/deploy/pages/deployment-option/cloud/vpc-byo-gcp.adoc

```diff
@@ -0,0 +1,524 @@
++
+```unset
+gcloud compute networks subnets create <primary-subnet-name> \
+    --project <host-project-id> \
+    --network <shared-vpc-name> \
+    --range 10.0.0.0/24 \
+    --region <region> \
+    --secondary-range <secondary-ipv4-range-name-for-pods>=10.0.8.0/21,<secondary-ipv4-range-name-for-pods>=10.0.1.0/24
+```
+
+. Get IAM permissions at the host project. 
++
+NOTE: Skip step if the host and service project are in the same GCP project.
++
+.. Get IAM permission of the primary subnet, and save the output etag:
++
+```unset
+gcloud compute networks subnets get-iam-policy <primary-subnet-name> \
+   --project <host-project-id> \
+   --region <region>
+```
++
+.. Get the project number (`PROJECT_NUMBER` column) of the service project:
++
+```unset
+gcloud projects list
+
+PROJECT_ID                      NAME                            PROJECT_NUMBER
+rp-byo-host                  rp-byo-service                     1030959231549
+rp-byoc-service              rp-byoc-host                       268950748084
+```
++
+.. Create the policy file with the content:
++
+```unset
+bindings:
+- members:
+  - serviceAccount:<service-project-number>@cloudservices.gserviceaccount.com
+  - serviceAccount:service-<service-project-number>@container-engine-robot.iam.gserviceaccount.com
+  role: roles/compute.networkUser
+etag: <etag>
+```
++
+.. Apply the policy saved to the above file:
++
+```unset
+gcloud compute networks subnets set-iam-policy <primary-subnet-name>  \
+    --project <host-project-id> \
+    --region <region> <policy-file-created-above>
+```
++
+.. Grant the host service agent the user role to the service account:
++
+```unset
+- gcloud projects add-iam-policy-binding <host-project-id> \
+    --member serviceAccount:service-<service-project-number>@container-engine-robot.iam.gserviceaccount.com \
+    --role roles/container.hostServiceAgentUser
+```
++
+.. Grant service project's GKE SA to manage firewall resources:
++
+```unset
+gcloud projects add-iam-policy-binding <host-project-id> \
+    --member=serviceAccount:service-<service-project-number>@container-engine-robot.iam.gserviceaccount.com \
+    --role=roles/compute.securityAdmin
+```
+For more information, see the https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc#managing_firewall_resources[Google documentation^].
+
+. Create a cloud router and NAT at the host project. For example:
++
+```unset
+gcloud compute routers create <route-name> \
+   --project <host-project-id> \
+   --region <region> \
+   --network <shared-vpc-name>
+
+gcloud compute addresses create <address-name>
+
+gcloud compute routers nats create <nat-config-name> \
+   --project <host-project-id> \
+   --router <router-name> \
+   --region <region> \
+   --nat-all-subnet-ip-ranges \
+   --nat-external-ip-pool <address-name> \
+   --enable-endpoint-independent-mapping
+```
+
+. Create VPC firewall rules.
```

Comment:
From here on down we seem to stop providing gcloud commands for most things. I worry that with such a long document and different formatting the customer will gloss over important bits that they need to do. I'm wondering if we can make the whole thing more uniform so it's very clear like "THIS SECTION IS A THING THAT YOU NEED TO DO" ha ha

We can provide more gcloud commands for things that were missing them.

Another idea to help the user navigate is to possibly split this page into multiple pages, one for creating a shared VPC all on it's own page, because that work is optional.